src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/wiremock/NomisApiMockServer.kt

```diff
@@ -48,18 +48,51 @@ class NomisApiMockServer : WireMockServer(WIREMOCK_PORT) {
   fun stubGetVisitsInitialCount(totalElements: Long) {
     NomisApiExtension.nomisApi.stubFor(
       get(
-        WireMock.urlPathEqualTo("/visits/ids")
-      ).willReturn(
-        aResponse()
-          .withHeader("Content-Type", "application/json")
-          .withStatus(HttpStatus.OK.value())
-          .withBody(visitPagedResponse(totalElements = totalElements))
+        urlPathEqualTo("/visits/ids")
       )
+        .withQueryParam("page", equalTo("0"))
+        .withQueryParam("size", equalTo("1"))
+        .willReturn(
+          aResponse().withHeader("Content-Type", "application/json").withStatus(HttpStatus.OK.value())
+            .withBody(visitPagedResponse(totalElements = totalElements))
+        )
     )
   }
+
+  fun stubMultipleGetVisitsCounts(totalElements: Long, pageSize: Long) {
+    // for each page create a response for each VisitId starting from 1 up to `totalElements`
```

Comment:
hideously complex to mock multiple request with slightly different parameters and responses.
Not sure if this is just too complex, I might rethink